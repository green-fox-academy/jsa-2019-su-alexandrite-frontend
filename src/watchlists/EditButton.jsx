import React from 'react';
import { MaterialIcons } from '@expo/vector-icons';
import { View, TouchableHighlight } from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

const EditButton = ({ isInEditMode, toggleEditMode }) => (
  <View style={styles.editor}>
    <TouchableHighlight style={styles.editorButton} onPress={() => toggleEditMode(!isInEditMode)} underlayColor="#eee">
      <MaterialIcons size={16} color="#666" name="edit" />
    </TouchableHighlight>
  </View>
);

EditButton.propTypes = {
  isInEditMode: PropTypes.bool.isRequired,
  toggleEditMode: PropTypes.func.isRequired,
};

export default EditButton;
