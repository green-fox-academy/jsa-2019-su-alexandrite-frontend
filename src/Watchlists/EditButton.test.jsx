import React from 'react';
import renderer from 'react-test-renderer';

import EditButton from './EditButton';

it('renders correctly', () => {
  const tree = renderer.create(<EditButton isInEditMode toggleEditMode={() => null} />).toJSON();
  expect(tree).toMatchSnapshot();
});
