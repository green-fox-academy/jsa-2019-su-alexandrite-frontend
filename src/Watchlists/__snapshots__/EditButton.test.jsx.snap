// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "flex-end",
      "flex": 1,
      "marginTop": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 6,
        "paddingHorizontal": 10,
        "paddingVertical": 8,
      }
    }
  >
    <Text />
  </View>
</View>
`;
